src/Symfony/Component/Validator/Constraints/FileValidator.php

```diff
@@ -51,6 +52,10 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($value instanceof UploadedFileInterface) {
+            $value = new UploadedFile('', $value->getClientFilename(), $value->getClientMediaType(), $value->getSize(), $value->getError());
```

Comment:
If this approach is followed, wouldn't it be better to use the bridge instead of duplicating the logic?

But I agree with @stof, a validator without dependency to HttpFoundation would be better (better performance, less complexity and dependencies).